README.md

```diff
@@ -68,3 +73,48 @@ You can find the greentea-client example application [here](./example/main.cpp).
     cmake --build cmake_build
 
 This generates an executable `./cmake_build/example/greentea-client-example`.
+
+# Adding greentea-client to a project
+
+## Build support
+
+greentea-client supports the CMake build system. To add greentea-client to a project, call
+[`add_subdirectory`](https://cmake.org/cmake/help/latest/command/add_subdirectory.html)
+pointing to the path to the client. Then link `greentea-client` to a CMake target that requires it with
+[`target_link_libraries`](https://cmake.org/cmake/help/latest/command/target_link_libraries.html).
+
+## Stream of I/O
+
+In order to communicate with a host, a stream of input/ouput (I/O) needs to be available to
+greentea-client.
+
+### stdio
+
+By default, greentea-client uses `<cstdio>` (`stdio.h`) from the C standard library (libc).
+On a PC, this normally points to the terminal's input and output. On embedded systems,
+some libc implementations support retargeting system I/O functions (e.g. to use a serial port).
+For example,
+* Arm Compiler 6: [Redefining target-dependent system I/O functions in the C library](https://developer.arm.com/documentation/100073/0615/The-Arm-C-and-C---Libraries/Redefining-target-dependent-system-I-O-functions-in-the-C-library?lang=en)
+* newlib (included in the GNU Arm Embedded Toolchain): [Reentrant system calls](https://sourceware.org/git/?p=newlib-cygwin.git;a=blob;f=newlib/libc/include/reent.h;h=2b01fbe8f329860fc7d76b681ea063375cb7eb72;hb=415fdd4279b85eeec9d54775ce13c5c412451e08#l13)
+
+### Alternative I/O
+
+If stdio retargeting is not available or a project needs to use a different stream of I/O
+from regular printing, an option is providing a custom implementation of the Greentea
+I/O functions declared in [`test_io.h`](./include/greentea-client/test_io.h). In this case
+the CMake definitions need to be adjusted accordingly:
+
+```cmake
+# Disable the default stdio-based implementation
+# before adding greentea-client
+SET(GREENTEA_CLIENT_STDIO OFF)
+
+# Add greentea-client
+add_subdirectory(greentea-client)
+
+# Add custom I/O implementation
+target_sources(greentea-client
+    PRIVATE
+        custom_test_io.cpp
+)
+```
```

Comment:
Could we make a custom IO example (our second example) and link to it instead of adding untested, may-go-stale code in the README?